src/main/java/org/jenkinsci/plugins/workflow/cps/CpsThreadGroup.java

```diff
@@ -391,6 +393,10 @@ private void run() throws IOException {
         if (doneSomeWork) {
             saveProgram();
         }
+        if (ending) {
+            execution.cleanUpHeap();
+            scripts.clear();
```

Comment:
Clearing `scripts` (introduced in #29) was also necessary, or even with the fix to the timing of `SerializableClassRegistry.release`, the test would still fail on Jenkins 1. The failure showed no root references, so I had to guess at the cause.